autorest/azure/async.go

```diff
@@ -81,10 +81,37 @@ func (f *Future) Done(sender autorest.Sender) (bool, error) {
 
 	resp, err := sender.Do(f.req)
 	f.resp = resp
-	if err != nil || !autorest.ResponseHasStatusCode(resp, pollingCodes[:]...) {
+	if err != nil {
 		return false, err
 	}
 
+	if !autorest.ResponseHasStatusCode(resp, pollingCodes[:]...) {
+		// check response body for error content
+		if resp.Body != nil {
+			type respErr struct {
```

Comment:
is this just so we can add the json tag to the field ?

Reply:
That's correct.  The body should look like this.
```json
{
  "error": {
    "code": "foo",
    "message": "bar"
  }
}
```